### PR TITLE
Tiny fix for errors not being caught in set_extent

### PR DESCRIPTION
**Pull request by corinnebosley**
Sometimes when plotting extents are very close to the poles, they are not being plotted due to some numerical issues with the geometries.  A fix had been implemented for this, but for some reason one of the conditions to try the workaround was not being met (that is, when crs was equal to self.projection, it was still being evaluated as False) so the workaround was being skipped.

This alteration should catch those conditions and apply the workaround.

### CODE REVIEW

File: lib/cartopy/mpl/geoaxes.py [hunk 1]
```diff
@@ -642,7 +641,7 @@ def set_extent(self, extents, crs=None):
         # extents are obviously the same as the projection domain.
         try_workaround = ((crs is None and
                            isinstance(self.projection, ccrs.PlateCarree)) or
-                          crs == self.projection)
+                          isinstance(crs, ccrs.PlateCarree))
```
Comment (dkillick) [inline lib/cartopy/mpl/geoaxes.py]: What if `crs` _isn't_ PlateCarree?

File: lib/cartopy/mpl/geoaxes.py [hunk 2]
```diff
@@ -1112,12 +1112,12 @@ def imshow(self, img, *args, **kwargs):
...
-        # clip the image. This does not work as the patch moves with mouse
-        # movement, but the clip path doesn't
-        # This could definitely be fixed in matplotlib
-#        if result.get_clip_path() in [None, self.patch]:
-#            # image does not already have clipping set, clip to axes patch
-#            result.set_clip_path(self.outline_patch)
+            # clip the image. This does not work as the patch moves with mouse
+            # movement, but the clip path doesn't
+            # This could definitely be fixed in matplotlib
+        #        if result.get_clip_path() in [None, self.patch]:
```
Comment (dkillick) [inline lib/cartopy/mpl/geoaxes.py]: I have to say, this rendering of the diff suggests that these comment characters still aren't indented correctly...

File: lib/cartopy/mpl/geoaxes.py [hunk 3]
```diff
@@ -1443,7 +1443,7 @@ def _pcolormesh_patched(self, *args, **kwargs):
         collection.set_alpha(alpha)
         collection.set_array(C)
         if norm is not None:
-            assert(isinstance(norm, mcolors.Normalize))
+            assert (isinstance(norm, mcolors.Normalize))
```
Comment (dkillick) [inline lib/cartopy/mpl/geoaxes.py]: I don't think this line needs the outer brackets at all. There's no line continuation or explicit operation order going on here...

Comment (corinnebosley) [reply lib/cartopy/mpl/geoaxes.py]: I actually didn't touch that... I'm confused...

File: lib/cartopy/mpl/geoaxes.py (hunk 2 above)
Comment (corinnebosley) [reply lib/cartopy/mpl/geoaxes.py]: I also didn't touch that... what's happening?

File: lib/cartopy/mpl/geoaxes.py (hunk 1 above)
Comment (corinnebosley) [reply lib/cartopy/mpl/geoaxes.py]: Yeah good point.  Didn't think that through, sorry.  I'll try again.

File: lib/cartopy/mpl/geoaxes.py (hunk 3 above)
Comment (dkillick) [reply lib/cartopy/mpl/geoaxes.py]: Has your editor gained sentience??

File: lib/cartopy/mpl/geoaxes.py (hunk 1 above)
Comment (ajdawson) [inline lib/cartopy/mpl/geoaxes.py]: This doesn't check the same thing does it? It was supposed to be checking if the given crs is the same as the projection (i.e. native coordinates, don't care what the projection actually is). Now it is just checking if the crs is `PlateCarree`, which seems wouldn't be useful unless we also knew the projection was `PlateCarree`. Can you explain the motivation here?

Comment (dkillick) [reply lib/cartopy/mpl/geoaxes.py]: @ajdawson the motivation is that according to @corinnebosley the original is not working...

Comment (ajdawson) [reply lib/cartopy/mpl/geoaxes.py]: That doesn't help at all. What I want to know is why are we changing the logic (i.e. the question we are asking) instead of the implementation of the logic?

It seems like the question we want the answer to is "is the specified crs the same as our projection?". Instead this change checks if the crs is `PlateCarree`, but why? How does this help determine if the workaround is needed? What if the specified crs matches the projection but isn't `PlateCarree`?

Comment (corinnebosley) [reply lib/cartopy/mpl/geoaxes.py]: Yes thank you @ajdawson, Pete already pointed out that observation.  It was an oversight on my part which I am now fixing.  I do still want to test that the crs is the same as the projection no matter what they are.

Comment (ajdawson) [reply lib/cartopy/mpl/geoaxes.py]: OK, timing difference between starting comment and posting, PR had 0 comments when I began.

File: lib/cartopy/mpl/geoaxes.py (hunk 3 above)
Comment (corinnebosley) [reply lib/cartopy/mpl/geoaxes.py]: I think it might have done.  It's certainly started being a bit stroppy and irrational lately...

**Comment by corinnebosley** [comment]
@dkillick This should be a bit more general as a check that the crs is the same as the projection.

I am having some major problems reproducing the failure mode though, so I can't get a new test to fail with the old behaviour (which would pass with the replacement behaviour).  Not sure what to do next...

**Comment by corinnebosley** [comment]
I discovered at 5pmish yesterday that the only way I could reproduce this problem was if the call to set_extent came from the ADAQ-Python code that I am working with, so I looked at that instead and found that some floats needed to be turned into ints, and that fixed it.

It still makes me a little nervous that the check I was trying to fix didn't work, when everything looked the same and the comparison should have returned True, but given the circumstances and the fact that I can't make a test to reproduce the problem, I'm just going to close this and consider it a teeny tiny corner case which will not happen again.
